app/controllers/api/v2/registration_tokens_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V2
+    class RegistrationTokensController < V2::BaseController
+      include Foreman::Controller::UsersMixin
+
+      include Foreman::Controller::Parameters::User
+      before_action :authenticate, :only => [:invalidate_tokens]
+
+      def resource_class
+        User
+      end
+
+      def find_resource(permission = :view_users)
+        editing_self? ? User.find(User.current.id) : User.authorized(permission).except_hidden.find(params[:id])
```

Comment:
@ekohl what do  you think, should we block changes to hidden accounts here?

Reply:
I was also wondering about that.